posts/map_blocks_example.md

```diff
@@ -0,0 +1,179 @@
+---
+author: Matt Long
+date: 2021-4-22
+tags: xarray, dask, pop
+---
+
+# How to Use `xarray.map_blocks` within a Mixed-Layer Depth Calculation
+
+Within this example, we cover how to use `xarray.map_blocks` to calculate the mixed-layer depth within CESM POP model output.
+
+This calculation is "embarassingly parellel" such that each calculation is done within a single a column. The calculation ***should*** be easily computed within each column across the model domain. This is where `map_blocks` can be used to improve the performance of this metric.
+
+## Imports and Data Ingestion
+
+We use the following libraries/packages within this example
+```python
+import os
+
+from scipy import interpolate
+import numpy as np
+import xarray as xr
+
+import matplotlib.pyplot as plt
+
+import pop_tools
+```
+
+Next, we read in the data using `xarray` and subset for temperature and salinity, subsetting for a smaller portion of the domain for testing purposes
+
```

Comment:
This post might be more powerful if we applied to the full dataset, demonstrating the scaling, etc.

Also, we might consider retaining/curating datasets like this for testing/demo purposes. If we have a collection of datasets like this, we can provide people with working examples. 

I noticed that these particular datasets are unwieldy because somewhere along the way in their generation, `xarray` put the `member_id` and perhaps the `time` dimension into the grid variables (i.e., `TAREA`, `KMT`, etc.). Super annoying.

Reply:
I think keeping it shorter for this post might be okay with the subset - in terms of executable notebooks, it might be helpful to contribute to the [NCAR Pangeo Gallery](https://gallery.pangeo.io/repos/NCAR/notebook-gallery/notebooks/Run-Anywhere/Ocean-Heat-Content/OHC_tutorial.html)? That way we can help increase the presence there instead of having our own separate gallery.